src/chief-v2.sol

```diff
@@ -0,0 +1,86 @@
+pragma solidity >=0.5.0;
+
+import "ds-note/note.sol";
+import "ds-math/math.sol";
+
+contract TokenLike {
+    function transferFrom(address, address, uint) public returns (bool);
+}
+
+contract Chief is DSNote, DSMath {
+    // --- Init ---
+    constructor(address governanceToken_) public { 
+        governanceToken = TokenLike(governanceToken_); 
+        threshold = WAD / 2;
+    }
+
+    // --- Data ---
+    uint256 public threshold; // locked % needed to enact a proposal
+    TokenLike public governanceToken; 
+    uint256 public locked; // total locked governanceToken
+    mapping(bytes32 =>    bool) public hasFired; // proposal => hasFired?
+    mapping(address => uint256) public balances; // guy => lockedGovHowMuch
+    mapping(address => bytes32) public picks; // guy => proposalHash
+    mapping(bytes32 => uint256) public votes; // proposalHash => votesHowMany
+
+    // --- Events ---
+    event Voted(
+        bytes32 indexed proposalHash,
+        address indexed voter, 
+        uint256 weight
+    );
+    event Executed(
+        address caller,
+        bytes32 proposal,
+        address indexed app,
+        bytes data
+    );
+
+    // --- Voting Interface ---
+    function lock(uint256 wad) public note {
+        require(governanceToken.transferFrom(msg.sender, address(this), wad), "ds-chief-transfer-failed");
+        balances[msg.sender] = add(balances[msg.sender], wad);
+        locked = add(locked, wad);
+
+        bytes32 currPick = picks[msg.sender];
+        if (currPick != bytes32(0) && !hasFired[currPick])
+            votes[currPick] = add(votes[currPick], wad);
+    }
+    function free(uint256 wad) public note {
+        balances[msg.sender] = sub(balances[msg.sender], wad);
+        require(governanceToken.transferFrom(address(this), msg.sender, wad), "ds-chief-transfer-failed");
+        locked = sub(locked, wad);
+
+        bytes32 currPick = picks[msg.sender];
+        if (currPick != bytes32(0) && !hasFired[currPick])
+            votes[currPick] = sub(votes[currPick], wad);
+    }
+
+    function vote(bytes32 currPick) public {
+        require(!hasFired[currPick], "ds-chief-proposal-has-already-been-enacted");        
+
+        uint256 weight   = balances[msg.sender];
+        bytes32 prevPick = picks[msg.sender];
+
+        if (prevPick != bytes32(0) && !hasFired[prevPick])
+            votes[prevPick] = sub(votes[prevPick], weight);
+
+        votes[currPick]   = add(votes[currPick], weight);
+        picks[msg.sender] = currPick;
+
+        emit Voted(currPick, msg.sender, weight);
+    }
+    function exec(address app, bytes memory data) public {
+        bytes32 proposal = keccak256(abi.encode(app, data));
+        require(!hasFired[proposal], "ds-chief-proposal-has-already-been-enacted");
+        require(votes[proposal] > wmul(locked, threshold), "ds-chief-proposal-does-not-pass-threshold");
```

Comment:
Can someone explain the math behind `wmul(locked, threshold)` and how it's equivalent to `votes[proposal] > locked / 2`?

How is it different from `votes[proposal] > wdiv(locked, 2 * WAD)`?

Reply:
When `threshold` is `0.5 WAD`, `wmul(locked, 0.5 WAD)` yields the same as `locked / 2` (or `wdiv(locked, 2 * WAD)`, as you say). `wmul` was chosen b/c it makes `threshold` represent the "percent of locked MKR needed to pass a proposal," which felt more natural than the alternatives. If we allow `threshold` to be a variable, we are giving MKR voters the ability to change the MKR % required to elect a proposal (possible w/ a [delegate call](https://github.com/dapphub/ds-chief/pull/7/files#diff-986b13d14e791c10decbbb6c214ee1dfR55)). You could argue that they should not have this power, the decision in this case was rather arbitrary.

Reply:
@xwvvvvwx do you think we should give chief the ability to alter its own "quorum" post-deployment?